vm/src/obj/objbyteinner.rs

```diff
@@ -472,6 +472,140 @@ impl PyByteInner {
         }
     }
 
+    pub fn delitem(
```

Comment:
Most of this code is taken from `list.__delitem__`, it could be deduplicated but I think this is a case of "duplicated code is better than the wrong abstraction", and atm we only have 2 copies of this basic code.